crates/sui/src/client_commands.rs

```diff
@@ -3021,3 +3177,43 @@ async fn execute_dry_run(
         .map_err(|e| anyhow!("Dry run failed: {e}"))?;
     Ok(SuiClientCommandResult::DryRun(response))
 }
+
+/// Call a dry run with the transaction data to get the estimated gas budget.
+/// The final gas budget is the dry run gas budget + 10% margin.
```

Comment:
The docs suggest that you're applying a 10% margin, but the implementation uses a fixed 1000 MIST overhead and also seems to be doing something with the computation cost and then the net cost. Can we replicate how this works in the TS SDK and/or wallet (maybe even with a commit and file pointer added to the doc comment here so others can see what it's based on), and then make sure the doc comment and the implementation match?

Reply:
Fixed, thanks!